lib/dotenv/rails.rb

```diff
@@ -27,15 +27,20 @@ def initialize
         logger: Dotenv::ReplayLogger.new,
         overwrite: false,
         files: [
-          root.join(".env.#{env}.local"),
-          (root.join(".env.local") unless env.test?),
-          root.join(".env.#{env}"),
-          root.join(".env")
+          ".env.#{env}.local",
+          (".env.local" unless env.test?),
+          ".env.#{env}",
+          ".env"
         ].compact,
         autorestore: env.test?
       )
     end
 
+    # The list of files to load, joined with Rails.root
+    def files
```

Comment:
@bkeepers fwiw this breaks the example from [Customizing Rails](https://github.com/bkeepers/dotenv?tab=readme-ov-file#customizing-rails) where we modify `Dotenv::Rails.files` in place.

From my current codebase, as we migrate from `dotenv-rails` to only use `dotenv`

```ruby
# This silently fails, and `.env.shared` is never loaded
Dotenv::Rails.files << ".env.shared" if defined?(Dotenv)

# this works
Dotenv::Rails.files += [".env.shared"] if defined?(Dotenv)
```